File: leetcode/0162.go
package leetcode

func findPeakElement(nums []int) int {
	l, r := 0, len(nums)-1
	for l < r {
		if mid := (r-l)/2 + l; nums[mid] <= nums[mid+1] {
			l = mid + 1
		} else {
			r = mid
		}
	}
	return l
}

File: leetcode/0162.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        int l = 0, r = nums.size() - 1;
        while (l < r) {
            if (auto mid = (r - l) / 2 + l; nums[mid] <= nums[mid + 1]) {
                l = mid + 1;
            } else {
                r = mid;
            }
        }
        return l;
    }
};